File: Data Structures & Algorithms/anagram-groups/submission-0.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string,unordered_map<char,int>> test(strs.size());
        vector<vector<string>> result;
        unordered_map<string,bool> picked;

        for (auto& str : strs){
            unordered_map<char,int> ana;
            for (auto& c : str){
                ana[c] += 1;
            }
            test[str] = ana;
            picked[str] = false;
        }

        for (int i = 0 ; i < strs.size(); i++){
            cout<<picked[strs[i]]<<endl;
            vector<string> res;
            if (!picked[strs[i]]) {
                res.push_back(strs[i]);
                picked[strs[i]] = true;
            }else{
                continue;
            }
            for(int j = i+1; j<strs.size(); j++){
                if(test[strs[i]] == test[strs[j]]){
                    res.push_back(strs[j]);
                    picked[strs[j]] = true;
                }
            }
            if(!res.empty()) result.push_back(res);
        }
        return result;
    }
};

File: Data Structures & Algorithms/reverse-integer/submission-0.py
class Solution:
    def reverse(self, x: int) -> int:
        res = 0
        check = False
        if x < 0:
            x *= -1
            check = True
        while x != 0:
            res *= 10
            if res > 2**31-1 or res < -2**31:
                return 0
            res += x%10
            x = x//10
        if check:
            res *= -1
        return res